components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -734,14 +739,28 @@ impl Interpolate for LengthOrNone {
                             let ia = fa.interpolate(&ta, time).unwrap();
                             result.push(TransformOperation::Rotate(fx, fy, fz, ia));
                         } else {
-                            // TODO(gw): Implement matrix decomposition and interpolation
-                            result.push(TransformOperation::Rotate(fx, fy, fz, fa));
+                            let matrix_f = rotate_to_matrix(fx, fy, fz, fa);
+                            let matrix_t = rotate_to_matrix(tx, ty, tz, ta);
+
+                            let decomposed_f = MatrixDecomposed2D::from(matrix_f);
+                            let decomposed_t = MatrixDecomposed2D::from(matrix_t);
+                            let interpolated = decomposed_f.interpolate(&decomposed_t, time).unwrap();
```

Comment:
this code is common (and will get more complicated when we handle 3d interpolation), perhaps we should implement `Interpolate` for `ComputedMatrix`?